crates/dpapi/src/crypto/mod.rs

```diff
@@ -88,9 +99,20 @@ pub fn cek_decrypt(
         });
     }
 
-    let kek = KekAes256::new(kek.into());
+    let key = Key::<Aes256Gcm>::try_from(kek).map_err(|_| CryptoError::InvalidKeyLength {
+        expected: Aes256Gcm::key_size(),
+        actual: kek.len(),
+    })?;
+    let kek = KekAes256::new(&key);
+
+    let blocks_len = key.len() / IV_LEN;
+    let blocks_len = blocks_len - 1;
+    let expected_len = blocks_len * IV_LEN;
+    let mut out_buf = vec![0; expected_len];
```

Comment:
```suggestion
    let mut out_buf = vec![0; key.len()];
```

This small patch fixes the problem mentioned above.

Reply:
Thank you Pavlo!